### ISSUE TITLE
Dune seems to be interpreting stanzas inside `.git`, creating problems when a branch is named `dune`

### ISSUE THREAD

**Issue by ejgallego** [ocaml/dune#1560]
In `master`, if we happen to be in a git checkout that has a branch named `dune`, dune will wrongly interpret this file as a `dune` file and fail:
```
File ".git/logs/refs/heads/dune", line 1, characters 0-40:
1 | 0000000000000000000000000000000000000000 ffaf01b3e49fb0ed9c8d1ffd3ae6da18e8e0da3c Emilio Jesus Gallego Arias <e+git@x80.org> 1542813227 +0100	branch: Created from refs/remotes/ejgallego/dune
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Unknown constructor 0000000000000000000000000000000000000000
```
it seems the `.git` directory should be treated as a data dir and thus `dune` files should not be interpreted there.

**Comment by diml** [comment]
That seems like a bug. Since #1517, the `.git` directory is seen by dune but is considered as a data directory, i.e. Dune shouldn't try to interpret files inside it.

**Comment by ejgallego** [comment]
Oh I see, indeed you want rules to be able to depend on `.git/HEAD` for example. So indeed there is a bug somewhere then.

**Comment by diml** [comment]
I pushed a reproduction case on master. The issue that when there is an `(ignored_subdirs ...)` stanza, we don't consider the default ignored sub-directories anymore.

One solution would be to implicitly prepend `:standard` to the argument of `ignored_subdirs`, however it defeats the purpose of #1517. BTW, @rgrinberg #1517 is effectively a breaking change as `(ignored_subdirs (blah))` is no longer accepted (dune complains about the first element of the list not being quoted). I feel like the simplest solution is to keep the syntax and semantic of `ignored_subdirs` unchanged, add a new stanza for the new semantic and get rid of `ignored_subdirs` in 2.0.

**Comment by ejgallego** [comment]
> The issue that when there is an `(ignored_subdirs ...)` stanza, we don't consider the default ignored sub-directories anymore.

Are you sure, I can see this problem even when I don't modify `(ignored_subdirs ...)`

**Comment by ejgallego** [comment]
Concretely, adding this trivial `dune` file makes the issue appear:
```
(library
 (name foo))
```

**Comment by diml** [comment]
Indeed, I extended the reproduction test with this case.

**Comment by rgrinberg** [comment]
> BTW, @rgrinberg #1517 is effectively a breaking change as (ignored_subdirs (blah)) is no longer accepted (dune complains about the first element of the list not being quoted)

I see. Sorry about that. I'm still puzzled why this is the case however. We are supposed to be parsing using the new syntax only in >= 1.6.

>  I feel like the simplest solution is to keep the syntax and semantic of ignored_subdirs unchanged, add a new stanza for the new semantic and get rid of ignored_subdirs in 2.0.

Okay, makes sense.

**Comment by ejgallego** [comment]
Fixed for me in 1.6
